cli/pcluster/config/pcluster_config.py

```diff
@@ -326,17 +326,27 @@ def __init_section_from_file(self, section_definition, config_parser, section_la
         except configparser.NoSectionError as e:
             self.error("Section '[{0}]' not found in the config file.".format(e.section))
 
+    def set_auto_refresh(self, refresh_enabled):
+        """Enable or disable refresh."""
+        self.__autorefresh = refresh_enabled
+
+    def auto_refresh(self):
+        """
+        Reload the sections structure and refresh all configuration sections and parameters.
+
+        This method is intended for internal refresh operations; configuration objects like Param, Section etc, must
+        call this method instead of refresh() to avoid unwanted side effects.
+        """
+        if self.__autorefresh:
+            self.refresh()
```

Comment:
should we make refresh method private?

Reply:
- refresh is to be called from external clients, so it should be public
- auto_refresh is meant for internal calls. This one could be private but needs to be called from param_types.

Reply:
The similarity of the two method names can actually be misleading. I'd rename ``auto_refresh`` to something like ``config_updated`` to clarify that it's a kind of event that has been raised.

Reply:
We could have opted for a full implementation of the Observer pattern, so that PclusterConfig would have been a decoupled observer of its sections and parameters but I think those classes are so strictly related that calling directly a PclusterConfig method from a Section Object is fine. Maybe we could prefix it with a single underscore to underline the fact that it's not a "public" method: ``_config_updated``.